include/alpaka/mem/view/ViewConst.hpp

```diff
@@ -31,11 +31,11 @@ namespace alpaka
             "This is not implemented"); // It might even be dangerous for ViewConst to store a reference to the wrapped
                                         // view, as this decouples the wrapped view's lifetime.
 
-        ALPAKA_FN_HOST_ACC ViewConst(const TView& view) : m_view(view)
+        ALPAKA_FN_HOST ViewConst(const TView& view) : m_view(view)
         {
         }
 
-        ALPAKA_FN_HOST_ACC ViewConst(TView&& view) : m_view(std::move(view))
+        ALPAKA_FN_HOST ViewConst(TView&& view) : m_view(std::move(view))
```

Comment:
Isn't this preventing to make a `ViewConst` inside the kernel? E.g. from a `ViewPlainPtr`.

Reply:
As I know we never supported the usage of views within a kernel. All buffer, Views are host objects.

Reply:
Buffers are definitely host objects. But `std::array` as well? Or a `T*` and a `size_t`?

Anyway, I am fine with the above changes. But they add restrictions.

Reply:
Yes `std::array` is a host-only object because you can only use the `constexpr` methods because runtime methods are not annotated with `__device__`.
`T*` does not require any annotations because it is not an object with operators.

Reply:
I was thinking about:
```c++
T* data = ...;
auto v = alpaka::createView(data, 10);
auto cv = alpaka::ViewConst{v};
```
inside kernel code. However, I just checked the API and we require to pass a device into `createView` and AFAIK devices are not allowed in kernel code. Therefore, you can proceed!